aiida/cmdline/utils/migration/v03_to_v04.py

```diff
@@ -0,0 +1,452 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida_core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Migration from v0.3 to v0.4, used by `verdi export migrate` command."""
+# pylint: disable=too-many-locals,too-many-statements,too-many-branches,unused-argument
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+
+import os
+
+import numpy as np
+
+from aiida.cmdline.utils import echo
+from aiida.cmdline.utils.migration.utils import verify_metadata_version, update_metadata
+
+__all__ = ('migrate_v3_to_v4',)
+
+
+def migrate_provenance_redesign(data):
+    """Utility function for `migrate_v3_to_v4` to handle REV. 1.0.20 (provenance redesign)"""
```

Comment:
is this a reference to a django migration?
might be helpful to add 1-2 words so that people know where to look for the specification of this.

Reply:
Yeah. We have done this throughout the actual migration function.
It is easier to reference the django migrations than the sqla, since django's migrations are naturally numbered.

I will add some extra explanation of this in the file's documentation, i.e. in the top of the file, where it at the moment "only" states what the general contents of this particular file is and where it is mainly used.

Reply:
Done